### PR TITLE
don't force stream destroy on shard_count update

### PR DESCRIPTION
**Pull request by nevins-b**
Addresses #521

### CODE REVIEW

**Comment by stack72** [comment]
Hi @nevins-b 

Thanks for the work here - please could you add an acceptance test to show the new functionality in action? As part of the acceptance test, it would be good to see that the kinesis stream didn't actually get recreated

Thanks

Paul

File: aws/resource_aws_kinesis_stream.go [hunk 1]
```diff
@@ -243,6 +245,35 @@ func setKinesisRetentionPeriod(conn *kinesis.Kinesis, d *schema.ResourceData) er
 	return nil
 }
 
+func updateKinesisShardCount(conn *kinesis.Kinesis, d *schema.ResourceData) error {
+	sn := d.Get("name").(string)
+
+	oraw, nraw := d.GetChange("shard_count")
+	o := oraw.(int64)
```
Comment (stack72) [inline aws/resource_aws_kinesis_stream.go]: You can't case to int64, it needs to be int

File: aws/resource_aws_kinesis_stream.go [hunk 2]
```diff
@@ -243,6 +245,35 @@ func setKinesisRetentionPeriod(conn *kinesis.Kinesis, d *schema.ResourceData) er
...
+
+	if n == o {
+		log.Printf("[DEBUG] Kinesis Stream (%q) Shard Count Not Changed", sn)
+		return nil
+	}
+
+	log.Printf("[DEBUG] Change %s Stream ShardCount to %d", sn, n)
+	_, err := conn.UpdateShardCount(&kinesis.UpdateShardCountInput{
+		StreamName:       aws.String(sn),
+		TargetShardCount: aws.Int64(n),
```
Comment (stack72) [inline aws/resource_aws_kinesis_stream.go]: needs to be:

```
TargetShardCount: aws.Int64(int64(n)),
```

**Comment by nevins-b** [comment]
Looks like it's working now, I forgot that you needed to look at open shards only

```
❯ make testacc TEST=./aws TESTARGS='-run=TestAccAWSKinesisStream_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSKinesisStream_ -timeout 120m
=== RUN   TestAccAWSKinesisStream_basic
--- PASS: TestAccAWSKinesisStream_basic (83.52s)
=== RUN   TestAccAWSKinesisStream_importBasic
--- PASS: TestAccAWSKinesisStream_importBasic (94.14s)
=== RUN   TestAccAWSKinesisStream_shardCount
--- PASS: TestAccAWSKinesisStream_shardCount (190.35s)
```

**Pull request merged**